ui/app/src/components/datasource/DatasourceList.tsx

```diff
@@ -59,45 +63,66 @@ export function DatasourceList<T extends Datasource>(props: DatasourceListProper
           project: getMetadataProject(datasource.metadata),
           name: datasource.metadata.name,
           displayName: getDatasourceDisplayName(datasource),
+          description: datasource.spec.display?.description,
+          type: datasource.spec.plugin.kind,
           version: datasource.metadata.version,
           createdAt: datasource.metadata.created_at,
           updatedAt: datasource.metadata.updated_at,
-          type: datasource.spec.plugin.kind,
         } as Row)
     );
   }, [data]);
 
   const [targetedDatasource, setTargetedDatasource] = useState<T>();
-  const [isEditDatasourceFormStateOpened, setEditDatasourceFormStateOpened] = useState<boolean>(false);
+  const [action, setAction] = useState<Action>('read');
+  const [isDatasourceDrawerOpened, setDatasourceDrawerOpened] = useState<boolean>(false);
+  const [isDeleteDatasourceDialogOpened, setDeleteDatasourceDialogOpened] = useState<boolean>(false);
 
   const handleDatasourceUpdate = useCallback(
     async (datasource: T) => {
       await onUpdate(datasource);
-      setEditDatasourceFormStateOpened(false);
+      setDatasourceDrawerOpened(false);
     },
     [onUpdate]
   );
 
   const handleRowClick = useCallback(
     (name: string, project?: string) => {
       setTargetedDatasource(findDatasource(name, project));
-      setEditDatasourceFormStateOpened(true);
+      setAction('read');
+      setDatasourceDrawerOpened(true);
+    },
+    [findDatasource]
+  );
+
+  const handleEditButtonClick = useCallback(
+    (name: string, project?: string) => () => {
+      const datasource = findDatasource(name, project);
+      setTargetedDatasource(datasource);
+      setAction('update');
+      setDatasourceDrawerOpened(true);
+    },
+    [findDatasource]
+  );
+
+  const handleDeleteButtonClick = useCallback(
+    (name: string, project?: string) => () => {
+      setTargetedDatasource(findDatasource(name, project));
+      setDeleteDatasourceDialogOpened(true);
     },
     [findDatasource]
   );
 
   const columns = useMemo<Array<GridColDef<Row>>>(
     () => [
       { field: 'project', headerName: 'Project', type: 'string', flex: 2, minWidth: 150 },
-      { field: 'type', headerName: 'Type', type: 'string', flex: 2 },
```

Comment:
I liked the fact that we see the type of the datasource. That's true that for now we are limited in the type of datasources but as soon as we'll have several, this will be important to know it.

Reply:
Actually it was not removed, but move to the same place as it's located for variables (= more to the left)

But later on I dont know if we should display the columns in the same order for both. E.g for datasources I'd like to bring back the type as first column, with a nice prometheus icon showm instead of "PrometheusDatasource" in full.

Reply:
Ah ! Sorry I didn´t see it. Fine to me. I agree that the text is a bit long. @Nexucis mentioned recently to give the possibility to put an icon alongside a plugin. That would be a good idea.